### PR TITLE
[SCAL-77263] Dynamic right nav implementation 

### PR DESCRIPTION
**Pull request by harsh-ts**
Task Link: https://thoughtspot.atlassian.net/browse/SCAL-77263

Screenshot:
image

### CODE REVIEW

File: docs/src/components/Docmap/index.tsx [hunk 1]
```diff
@@ -1,26 +1,35 @@
...
-    </div>
-);
+const Docmap = (props: { docContent: string }) => {
+
+    const [toc, setToc] = useState('');
+
+    useEffect(() => {
+        const doc = document.createElement('div');
+        doc.innerHTML = props.docContent;
+        const tocEl = doc.querySelector('#toc');
```
Comment (ayonghosh) [inline docs/src/components/Docmap/index.tsx:11]: So the toc is already present in the DOM and is hidden? Why can't we use CSS to make it visible?

**Review by ayonghosh — COMMENTED** [review]
The content seems to be all in bold in the screenshot, can we fix that?

**Review by pankajp-ts — APPROVED** [review]
After discussion with @harsh-ts , LGTM 👍

File: docs/src/assets/styles/index.scss [hunk 2]
```diff
@@ -71,7 +75,7 @@ a {
 
 .introWrapper {
     display: flex;
-    padding-top: 81px;
+    // padding-top: 81px;
```
Comment (ayonghosh) [inline docs/src/assets/styles/index.scss:78]: Let's remove this if not required

**Pull request merged**